server/service/integration_core_test.go

```diff
@@ -3649,9 +3793,13 @@ func (s *integrationTestSuite) TestLabels() {
 	assert.Len(t, summaryResp.Labels, builtInsCount+1)
 
 	// next page is empty
-	s.DoJSON("GET", "/api/latest/fleet/labels", nil, http.StatusOK, &listResp, "per_page", strconv.Itoa(builtInsCount+1), "page", "1", "query", t.Name())
```

Comment:
It seems we were returning `http.StatusOK` when providing a `"query"`? (If so, aren't we breaking the API? - even if the change makes sense)

Reply:
Yes? But anyone who would see the "break" would be being alerted to previously silent fact that they were not getting the results that they expected via the query param? Defer to you on whether or not to do it.

Reply:
Agree. Let's add a note to the `changes/` that this is a breaking change (to fix a bug) so that users are aware.

Reply:
@noahtalerman let me know if you are a-ok with the minor breaking change.

The other option is to (later / on another PR) add support for `"query"` on this endpoint?

Reply:
@lucasmrod here is [that additional change file](https://github.com/fleetdm/fleet/pull/17455/commits/40cebbe99a8ff01ce94f638f8c6b1c02a3d53e30)

Reply:
Thanks folks!

Tagging @rachaelshaw here. I think up to Rachael (API design DRI) on the breaking change.

Reply:
@rachaelshaw we actually do not document this param as usable with this verb / endpoint in the first place, so this "break" is even less of a "break."